packages/react-saasify/src/components/ServiceForm/ReactFromJSONSchema.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+
+import ReactFromJSON from 'react-from-json'
+
+const mapProp = (prop, onChange) => {
+  if (prop && prop !== onChange && prop.type) {
+    const { type, ...props } = prop
+
+    return {
+      type,
+      props: {
+        ...props,
+        onChange,
+        name: props.propKey
+      }
+    }
+  }
+
+  return prop
+}
+
+export const ReactFromJSONSchema = ({ definition, mapping, onChange }) => (
+  <ReactFromJSON
+    entry={{ ...definition.params.schema.properties, type: 'entry' }}
+    mapping={mapping}
+    mapProp={(prop) => mapProp(prop, onChange)}
```

Comment:
This inline function is a perf pitfall for react that will cause it to always re-render this subtree.

outside of func:
`const memMapProp =  mem((onChange) => (prop) => mapProp(prop. onChange))`

and use `memMapProp` here instead.

Reply:
Aha, I knew you'd pick up on this. I tend to sit in the "use arrow functions freely" camp, i.e. https://cdb.reacttraining.com/react-inline-functions-and-performance-bdff784f5578

Reply:
Yeah, I've trained my eyes to look for these cases and should prolly relax them a bit. Spent too much time focused on RN perf optimizations for clients lol.

It may be worth optimizing here since the `ReactFromJSON` subtree may be non-trivial.